services/api/kalvi/api/views/auth.py

```diff
@@ -155,19 +155,24 @@ def post(self, request, uid, token, format=None):
         except ValidationError as e:
             return Response({'error': e.detail}, status=status.HTTP_400_BAD_REQUEST)
         
-# Viewset class for blocking refresh tokens after logging out.       
+# Viewset class for blocking tokens after logging out.       
 class SignOutEndpoint(APIView):
     def post(self, request):
         refresh_token = request.data.get('refresh_token')
-        if refresh_token:
+        access_token = request.headers.get('Authorization').split(' ')[1] #We are catching access tokens from authorization header.
+        if refresh_token:  
             try:
                 # Connect to Redis
                 redis_conn = get_redis_connection()
-                token = str(RefreshToken(refresh_token))
-                # Blacklist the token in Redis
-                expiration_time = int(settings.SIMPLE_JWT['REFRESH_TOKEN_LIFETIME'].total_seconds())
-                redis_conn.set(token, 'blacklisted')
-                redis_conn.expire(token, expiration_time)
+                refresh_token_str = str(RefreshToken(refresh_token))
+                access_token_str = str(AccessToken(access_token))
+                # Blacklist tokens in Redis
+                refresh_exp_time = int(settings.SIMPLE_JWT['REFRESH_TOKEN_LIFETIME'].total_seconds())
+                access_exp_time = int(settings.SIMPLE_JWT['ACCESS_TOKEN_LIFETIME'].total_seconds())
+                with redis_conn.pipeline() as pipe:
```

Comment:
Create a separate class for Redis in a separate file and implement the logic there.

Reply:
ohkiee ohkiee